src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.Linux.cs

```diff
@@ -86,7 +86,20 @@ private int InternalBind(NetworkCredential tempCredential, SEC_WINNT_AUTH_IDENTI
             else
             {
                 string LDAP_SASL_SIMPLE = null;
-                error = Interop.Ldap.ldap_sasl_bind(_ldapHandle, cred.user, LDAP_SASL_SIMPLE, new berval(cred.password), IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+                berval password = new berval
+                {
+                    bv_len = cred.password.Length,
+                    bv_val = Marshal.StringToHGlobalAuto(cred.password),
```

Comment:
As written, I think the release is exception safe, since it is not possible for bv_val to be assigned, and yet an exception occur before the try block. But it might be cleaner to simply put it inside the try, in case someone adjusts the code later. FreeHGlobal accepts null wthout a problem.